modules/cluster/cloud-config.yml

```diff
@@ -1,112 +1,111 @@
-    permissions: "0744"
-    owner: "root"
-    content: |
-      ECS_CLUSTER=${ecs_cluster_name}
-      ECS_LOG_LEVEL=${ecs_log_level}
-      ECS_ENABLE_CONTAINER_METADATA=true
-      ECS_ENABLE_TASK_IAM_ROLE=true
-      ECS_ENABLE_TASK_IAM_ROLE_NETWORK_HOST=true
-      ECS_AVAILABLE_LOGGING_DRIVERS=["awslogs"]
+- path: "/etc/ecs/ecs.config"
+  permissions: "0744"
+  owner: "root"
+  content: |
+    ECS_CLUSTER=${ecs_cluster_name}
+    ECS_LOG_LEVEL=${ecs_log_level}
+    ECS_ENABLE_CONTAINER_METADATA=true
+    ECS_ENABLE_TASK_IAM_ROLE=true
+    ECS_ENABLE_TASK_IAM_ROLE_NETWORK_HOST=true
+    ECS_AVAILABLE_LOGGING_DRIVERS=["awslogs"]
 
-  - path: "/usr/local/scripts/cloudformation-signal.sh"
-    permissions: "0744"
-    owner: "root"
-    content: |
-      #! /bin/bash
-
-      set -euo pipefail
-
-      function await_process() {
-        echo -n "Waiting for $1..."
-        while ! pgrep -f "$1" > /dev/null; do
-          sleep 1
-        done
-        echo "Done!"
-      }
-      await_process "/usr/libexec/amazon-ecs-init start"
+- path: "/usr/local/scripts/cloudformation-signal.sh"
+  permissions: "0744"
+  owner: "root"
+  content: |
+    #! /bin/bash
+    set -uo pipefail
+    INSTANCE_ID=$(curl http://169.254.169.254/latest/meta-data/instance-id)
+    echo -n "Waiting for Instance $INSTANCE_ID to be ACTIVE in cluster ${ecs_cluster_name}"
+    function await_process() {
+      CHECK_ACTIVE=$(aws --region ${region} ecs list-container-instances --cluster ${ecs_cluster_name} --status ACTIVE --filter ec2InstanceId==$INSTANCE_ID | grep 'arn:aws:ecs')
+      while [ -z $CHECK_ACTIVE ]; do
+        sleep 1
+        CHECK_ACTIVE=$(aws --region ${region} ecs list-container-instances --cluster ${ecs_cluster_name} --status ACTIVE --filter ec2InstanceId==$INSTANCE_ID | grep 'arn:aws:ecs')
+      done
+      echo "Done!"
+      /opt/aws/bin/cfn-signal -e $? --stack ${stack_name} --resource AutoScalingGroup --region ${region}
+    }
+    await_process
 
 runcmd:
-  - |
-    yum install -y https://amazon-ssm-${region}.s3.amazonaws.com/latest/linux_amd64/amazon-ssm-agent.rpm
-  - |
-    mkdir -p /tmp/cloudwatchagent && cd /tmp/cloudwatchagent
-    curl -O https://s3.amazonaws.com/amazoncloudwatch-agent/linux/amd64/latest/AmazonCloudWatchAgent.zip
-    unzip AmazonCloudWatchAgent.zip && ./install.sh
-    rm -rf /tmp/cloudwatchagent
-    start amazon-cloudwatch-agent
-  - |
-    start ecs
-    /usr/local/scripts/cloudformation-signal.sh
-    /opt/aws/bin/cfn-signal -e $? --stack ${stack_name} --resource AutoScalingGroup --region ${region}
+- |
+  yum install -y https://s3.dualstack.${region}.amazonaws.com/amazon-ssm-${region}/latest/linux_amd64/amazon-ssm-agent.rpm
+  curl -O https://bootstrap.pypa.io/get-pip.py
+  python get-pip.py --user
+  export PATH=~/.local/bin:$PATH
+  pip install awscli --upgrade --user
+  systemctl enable awslogsd.service
+  systemctl start awslogsd
+  /usr/local/scripts/cloudformation-signal.sh > /dev/null 2>&1 &
```

Comment:
I don't think we should sink the output of this script, as it could contain useful information for debugging. Also, why are we running it in the background?

Reply:
It has to run in the background as ECS does not start until the User script has finished running.  Otherwise it just hangs waiting for the instance to join the cluster but it can't join the cluster because the user script hasn't finished.

I'll check where the output actually gets logged to if I remover the /dev/null - by the time this finishes executing standard out is probably not going to cloud-init-output.log

Reply:
> /dev/null 2>&1 removed - output is written to cloud-init-output.log